qiskit/algorithms/optimizers/spsa.py

```diff
@@ -63,7 +63,7 @@ class SPSA(Optimizer):
 
         SPSA can be used in the presence of noise, and it is therefore indicated in situations
         involving measurement uncertainty on a quantum computation when finding a minimum.
-        If you are executing a variational algorithm using a Quantum ASseMbly Language (QASM)
+        If you are executing a variational algorithm using an OpenQASM
```

Comment:
Is it relevant that OpenQASM might be used as an intermediary when running on a simulator? If not, it's probably better to omit "OpenQASM" altogether and just write "... executing a variational algorithm using a simulator or a real device ..."

Reply:
I think `qasm` here was originally more in the context of the `qasm_simulator` versus `statevector_simulator` where the former has shot noise i.e. `SPSA can be used in the presence of noise`

Reply:
That makes sense. If this is still valid, then there's no reason to remove "an OpenQASM"